examples/subscribe.cpp

```diff
@@ -0,0 +1,97 @@
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+#include <ews/ews.hpp>
+#include <ews/ews_test_support.hpp>
+
+#include <chrono>
+#include <thread>
+#include <cstdlib>
+#include <exception>
+#include <iostream>
+#include <ostream>
+#include <string>
+
+template <class... Ts> struct overloaded : Ts...
+{
+    using Ts::operator()...;
+};
+template <class... Ts> overloaded(Ts...)->overloaded<Ts...>;
+
+int main()
+{
+    int res = EXIT_SUCCESS;
+    ews::set_up();
+
+    try
+    {
+        const auto env = ews::test::environment();
+        auto service = ews::service(env.server_uri, env.domain, env.username,
+                                    env.password);
+
+        // Subscribe to all <CreatedEvents> in inbox
+        auto sub_info = service.subscribe({ews::standard_folder::inbox},
+                                          {ews::event_type::created_event}, 10);
+
+        // Create and send a message to trigger an event
+        auto message = ews::message();
+        message.set_to_recipients({ews::mailbox("test2@otris.de")});
+        service.create_item(message,
+                            ews::message_disposition::send_and_save_copy);
+
+        // Wait for the event to be created
+        std::this_thread::sleep_for(std::chrono::seconds(5));
+
+        // Get and check the created_event.
+        // This way, we can get all the needed information from the events for
+        // further handling.
+        auto notification = service.get_events(sub_info.get_subscription_id(),
+                                         sub_info.get_watermark());
+        std::cout << "SubscriptionId: " << notification.subscription_id
+                  << std::endl;
+        std::cout << "MoreEvents: " << notification.more_events << std::endl;
+        for (auto& e : notification.events)
+        {
+            std::visit(
+                overloaded{
+                    [](auto&& arg) {
+                        std::cout << "EventType: "
+                                  << ews::internal::enum_to_str(arg.get_type())
+                                  << std::endl;
+                        std::cout << "Watermark: " << arg.get_watermark()
+                                  << std::endl;
+                        std::cout << "Timestamp: " << arg.get_timestamp()
+                                  << std::endl;
+                    },
+                    [](ews::status_event arg) {
+                        std::cout << "EventType: "
+                                  << ews::internal::enum_to_str(arg.get_type())
+                                  << std::endl;
+                        std::cout << "Watermark: " << arg.get_watermark()
+                                  << std::endl;
+                    }},
```

Comment:
Just handle `created_event` and use an empty `[](auto&&){ /* do nothing */ }` lambda for the rest? Example gets more concise this way.